clustered/integration-test/src/test/java/org/ehcache/clustered/BasicCacheOpsMultiThreadedTest.java

```diff
@@ -181,11 +181,17 @@ private static PersistentCacheManager createCacheManager(URI clusterURI) {
     return clusteredCacheManagerBuilder.build(false);
   }
 
-  private void silentPuts(Cache<Long, String> customValueCache) {
-    customValueCache.put(2L, "value2");
-    customValueCache.put(2L, "value21");
-    customValueCache.put(3L, "value3");
-    customValueCache.put(4L, "value4");
-    customValueCache.put(2L, "value22");
+  private void parallelPuts(Cache<Long, String> customValueCache) {
```

Comment:
These changes were put to reproduce locally..

I was able to reproduce the problem only when I put an explicit random sleep inside the connected() method which cannot be raised in a pull request..

But I have manually verified the issue is fixed this time by verifying it with a random sleep inside the connected() method, which was then reverted before raising this pull request.